kosat-core/src/commonMain/kotlin/org/kosat/Restarter.kt

```diff
@@ -26,16 +26,15 @@ class Restarter(private val solver: CDCL) {
 
     private var lubyPosition = 1
 
-    fun restart() {
+    private fun restart() {
         restartNumber = lubyMultiplierConstant * luby(lubyPosition++)
         solver.backtrack(0)
     }
 
-    fun update() {
-        numberOfConflictsAfterRestart++
+    fun restartIfNeeded() {
         if (numberOfConflictsAfterRestart >= restartNumber) {
-            numberOfConflictsAfterRestart = 0
             restart()
+            numberOfConflictsAfterRestart = 0
```

Comment:
Probably this should be handled not inside the `Restarter`, but via a budget in `search(int numConfls)` method...